self-core-impl/src/main/java/com/selfxdsd/core/GitlabWebhooks.java

```diff
@@ -0,0 +1,114 @@
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.core;
+
+import com.selfxdsd.api.Project;
+import com.selfxdsd.api.Webhooks;
+import com.selfxdsd.api.storage.Storage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.json.Json;
+import java.net.HttpURLConnection;
+import java.net.URI;
+
+/**
+ * Gitlab repo webhooks.
+ *
+ * @author criske
+ * @version $Id$
+ * @since 0.0.13
+ */
+final class GitlabWebhooks implements Webhooks {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+            GitlabWebhooks.class
+    );
+
+    /**
+     * Gitlab repo Webhooks base uri.
+     */
+    private final URI hooksUri;
+
+    /**
+     * Gitlab's JSON Resources.
+     */
+    private final JsonResources resources;
+
+    /**
+     * Self storage, in case we want to store something.
+     */
+    private final Storage storage;
+
+    /**
+     * Ctor.
+     *
+     * @param resources Gitlab's JSON Resources.
+     * @param hooksUri Hooks base URI.
+     * @param storage Storage.
+     */
+    GitlabWebhooks(final JsonResources resources,
+                   final URI hooksUri,
+                   final Storage storage) {
+        this.resources = resources;
+        this.hooksUri = hooksUri;
+        this.storage = storage;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <br/>
+     * More about Gitlab project webhook configuration
+     * <a href="https://docs.gitlab.com/ee/api/projects.html#add-project-hook">
+     * here</a>.
+     */
+    @Override
+    public boolean add(final Project project) {
+        LOG.debug("Adding Gitlab webhook for Project "
+                + project.repoFullName());
+        final boolean added;
+        final Resource response = this.resources.post(
+                this.hooksUri,
+                Json.createObjectBuilder()
+                        .add("id", project.repoFullName()
+                                .replace("/", "%2F"))
+                        .add("url", "https://self-xdsd.com/gitlab/"
+                                + project.repoFullName())
+                        .add("issues_events", true)
+                        .add("token", project.webHookToken())
+                        .build()
+        );
+        if (response.statusCode() == HttpURLConnection.HTTP_CREATED) {
+            added = true;
+            LOG.debug("Webhook added successfully!");
+        } else {
+            added = false;
+            LOG.debug("Problem when adding webhook. Expected 201 CREATED, "
+                            + " but got " + response.statusCode());
+        }
+        return added;
```

Comment:
@criske same in this method, indentation seems broken

Reply:
@amihaiemil Sometimes my Intellij really is messing up the indentation. eventhough I'm using reformat-code and auto-indent commands. Just updated to the latest version maybe that will fix it.